spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Linear.scala

```diff
@@ -136,16 +135,15 @@ class Linear[T: ClassTag](
       gradBias.resize(outputSize)
     }
 
-    val value = ev.fromType[Double](scale)
     if (input.dim() == 1) {
-      gradWeight.addr(value, gradOutput, input)
+      gradWeight.addr(ev.fromType[Double](scaleW), gradOutput, input)
       if (withBias) {
-        gradBias.add(value, gradOutput)
+        gradBias.add(ev.fromType[Double](scaleB), gradOutput)
       }
     }
     else if (input.dim() == 2) {
-      gradWeight.addmm(value, gradOutput.t, input)
-      if (withBias) gradBias.addmv(value, gradOutput.t, addBuffer)
+      gradWeight.addmm(ev.fromType[Double](scaleW), gradOutput.t, input)
+      if (withBias) gradBias.addmv(ev.fromType[Double](scaleB), gradOutput.t, addBuffer)
```

Comment:
This looks not right. We should scale the gradient after the regularizer process the gradient. @psyyz10 . Also please check other place where regularizer is used.